src/core/public/chrome/ui/header/right_navigation_button.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiHeaderSectionItemButton, EuiIcon } from '@elastic/eui';
+import React from 'react';
+import { CoreStart } from 'src/core/public';
+
+export interface RightNavigationButtonProps {
+  application: CoreStart['application'];
+  http: CoreStart['http'];
+  appId: string;
+  iconType: string;
+  title: string;
+}
+
+export const RightNavigationButton = ({
+  application,
+  http,
+  appId,
+  iconType,
+  title,
+}: RightNavigationButtonProps) => {
+  const navigateToApp = () => {
+    const appUrl = application.getUrlForApp(appId, {
+      path: '/',
+      absolute: false,
+    });
+    // Remove prefix in Url including workspace and other prefix
+    const targetUrl = http.basePath.prepend(http.basePath.remove(appUrl), {
+      withoutClientBasePath: true,
+    });
```

Comment:
I am thinking if we just use `navigateToApp` here, the method will detect if the app is accessible inside workspace and use the correct prepend params when navigate.

Reply:
Here we use `application.navigateToUrl`. Does this also be covered?

Reply:
Nope, navigateToUrl won't detect app related info.

Reply:
Oh, yes. navigateToUrl will parse the app info from the url. Forget it.